CrossCorrelate and GetDetectorOffset algorithm user docs (#38538)
mantidproject/mantid: docs/source/algorithms/GetDetectorOffsets-v1.rst

Description
-----------

> This algorithm requires a workspace where the x-array contains offsets, in number of bins, generated by the :ref:`algm-CrossCorrelate` algorithm.

Reviewer: ```suggestion
This algorithm requires a workspace containing a cross correlation, generated by the :ref:`algm-CrossCorrelate` algorithm which should contain a single peak and where the x-array contains offsets, in number of bins.
```

In this first step you select one spectrum to be the
reference spectrum and all of the other spectrum are cross correlated
against it. Each output spectrum then contains a peak whose location

> defines the offset from the reference spectrum.

Reviewer: These lines could be deleted if we go with my proposed edit for line 12 above

> The algorithm iterates over each spectrum in the workspace and fits a PeakFunction (default is a :ref:`Gaussian <func-Gaussian>` function) with :ref:`linear background <func-LinearBackground>` to the cross-correlation spectra.

Reviewer: ```suggestion
The algorithm iterates over each spectrum in the input workspace containing the cross-correlation spectra and fits a PeakFunction (default is a :ref:`Gaussian <func-Gaussian>` function) with :ref:`linear background <func-LinearBackground>`
```

:ref:`algm-CreateCalFileByNames` algorithm can be used to
alter the grouping in the cal file.

Estimating the CC-spectrum fit parameters
#########################################

The starting value for the fit is that the linear background is zero.

> The peak center is found in two steps. First is the x-value of the heighest point in the whole cross-correlation spectrum.

Reviewer: ```suggestion
The peak center is found in two steps. First is the x-value of the highest point in the whole cross-correlation spectrum.
```

Then it is observed again by subtracting the background (which is zero) then finding the heighest point again within the neighborhood of the original data.

> The height is the background subtracted y-value of the heightest point as described earlier.

Reviewer: ```suggestion
The height is the background subtracted y-value of the highest point as described earlier.
```

The full-width-half-maximum is :math:`2 \sigma \sqrt{2 ln(2)}` where :math:`\sigma=10` if ``EstimateFWHM=False``.
When ``True``, the value is estimated by integrating the background subtracted area under the window and dividing by the height.

**Problems can arise** for fitting if the cross-correlation peak is broad or the background is high and/or with significant slope.
Since a broad peak can be estimated as a very broad peak with a relatively high background, the peak can be easily misfit.
Similarly, since the starting value for the background is zero, a heavily sloping background will interfere with determining a starting value for the peak position.

> 

Reviewer: ```suggestion
Also note that if the input cross-correlation has been calculated over too large a d-spacing range it can contain "harmonics" in addition to the fundamental peak and this will obviously confuse the peak fitting algorithm.
```
